stl/src/filesystem_space.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// This must be as small as possible, because its contents are
+// injected into the msvcprt.lib and msvcprtd.lib import libraries.
+// Do not include or define anything else here.
+// In particular, basic_string must not be included here.
+
+// TRANSITION, the code in this file should be moved back to filesystem.cpp
+// when a Windows 10 SDK beyond version 1903 is available (see GH-322).
+
+#include <internal_shared.h>
+#include <limits.h>
+#include <xfilesystem_abi.h>
+
+#include <Windows.h>
+
+namespace {
+    static_assert(sizeof(uintmax_t) == sizeof(ULARGE_INTEGER) && alignof(uintmax_t) == alignof(ULARGE_INTEGER),
+        "Size and alignment must match for reinterpret_cast<PULARGE_INTEGER>");

```

Comment:
Curious, any reason for this `static_assert` instead of using `QuadPart` on `ULARGE_INTEGER`, `bit_cast`ing it, or using bit-shifts on `HighPart` and `LowPart`?

Reply:
1. Because `QuadPart` is not necessarily `intmax_t`.
2. Because `bit_cast` didn't exist when this code was written (and still does not exist).
3. Because the shifts etc. have a code size and maintainability cost.
4. Because we want the pointers to be passed down directly without this code needing to dereference them at all.

Reply:
I've filed #357 to track this. Note that the API wants to scribble into `ULARGE_INTEGER` variables which we don't have (all we have are `uintmax_t*`), so if we want to avoid the `reinterpret_cast`s, we would need to introduce such variables (at which point we can just assign from `QuadPart`).